db/migrate/20200809115958_add_role_to_user_workspaces_table.rb

```diff
@@ -0,0 +1,13 @@
+class AddRoleToUserWorkspacesTable < ActiveRecord::Migration[6.0]
+  def up
+    add_column :users_workspaces, :id, :primary_key
+    add_column :users_workspaces, :role, :integer, nil: false
```

Comment:
Do you think it's a bad idea to assign default role here as it was with user.role column?

I'm not quite understood that. You have method `add_user_to_workspace` in AssignUserService class and this method creates membership record:  `workspace.users << user`

But how it works if users_workspaces.role cannot be created with NULL value? Because you assign 'staff' role in `set_default_role_for_user` method that called after that.